src/emmy/value.cljc

```diff
@@ -92,13 +90,13 @@
   [x]
   #?(:clj
      (or (instance? Number x)
-         (instance? Complex x))
+         (core/= (kind x) :emmy.complex/complex))
      :cljs (or (cljs.core/number? x)
                (core/= "bigint" (goog/typeOf x))
                (instance? Fraction x)
                (instance? goog.math.Integer x)
                (instance? goog.math.Long x)
```

Comment:
is it significantly cheaper to check `(instance? emmy.complex.Complex x)`? just curious if kind is expensive here.

Reply:
I would have done that except you need to use the type in `impl` which felt like cheating. Let me see if the time difference is significant. If it is, then I agree that we should choose the quicker thing. Alternately, we could make the protocol INumerical handle this?

Reply:
```Clojure
(let [c (emmy.complex/complex 1 2)]
  (time
   (dotimes [_ 100000]
     (assert (= :emmy.complex/complex (emmy.value/kind c))))))
"Elapsed time: 8.2825 msecs"

(let [c (emmy.complex/complex 1 2)]
  (time
   (dotimes [_ 100000]
     (assert (instance? Complex c)))))
"Elapsed time: 1.419541 msecs"
```

Reply:
∴ fixing

Reply:
now we have the dependency problem again, which `kind` solved for us... maybe I'll leave it alone for now, and we could think of another way to do this

Reply:
I spent a little more time on this. Yes, `kind` is about 3x more expensive than `instance?` for this test. But we can't import `Complex` into `value.cljc`. There's one way around this: a marker interface. We could use something like `(instance? emmy.value.Numerical v)` where instead of Numerical having a method we call, we use it purely as a marker interface, as something that should be thought of as part of the numeric tower without being known to `value`. We could look at what  `Numerical` is used for and try to make `numerical?` equivalent to either being part of the known numeric tower or possessing the marker interface.

Reply:
OK: I added the marker interface INumericTower.